### PR TITLE
Add filter by plugin in the catalog

### LINKED ISSUE #4097: Add a "filter by plugin" in the Catalog view
If Kubeapps is configured with multiple plugins, a user may want to filter by the plugin type (eg, "list all the carvel packages").

Adding a new filter in the Catalog view (ie, just frontend) should be enough. Long-term, we might consider adding backed filters to the core API to query just a subset of the configured plugins.

### PR DESCRIPTION
**Pull request by antgamdia**
### Description of the change

This is a replacement of the old https://github.com/vmware-tanzu/kubeapps/pull/4100, but using the getConfiguredPlugins instead.
Basically, this is adding a new "filter by plugin" option in the catalog.

### Benefits

It will become easier to look up for pkgs in the catalog.

### Possible drawbacks

More complexity when it comes to filtering perhaps? 

### Applicable issues

- fixes #4097

### Additional information

image

### CODE REVIEW

File: dashboard/src/components/Catalog/Catalog.tsx [hunk 1]
```diff
@@ -235,7 +237,10 @@ export default function Catalog() {
     repos
       .filter(r => !r.namespaceScoped || r.packageRepoRef?.context?.namespace === namespace)
       .map(r => r.name),
-  );
+  ).sort();
+  const allPlugins = uniq(
+    configuredPlugins.filter(p => p.name.includes(".packages")).map(p => getPluginName(p) || ""),
```
Comment (castelblanque) [inline dashboard/src/components/Catalog/Catalog.tsx:242]: Is this `.packages` plugin name a convention?
Perhaps in the future the getConfiguredPlugins API could return a specific field determining the type of plugin.

File: dashboard/src/components/Catalog/Catalog.tsx [hunk 2]
```diff
@@ -483,6 +493,19 @@ export default function Catalog() {
                   />
                 </div>
               )}
+              {allPlugins.length > 0 && (
+                <div className="filter-section">
+                  <label className="filter-label">Package Type</label>
+                  <FilterGroup
+                    name={filterNames.PKG_TYPE}
+                    options={allPlugins}
```
Comment (castelblanque) [inline dashboard/src/components/Catalog/Catalog.tsx:501]: Perhaps we should differentiate the plugin name from the package type label. With this, we are listing a friendly name of active plugins, e.g.: 

- Helm
- Kapp
- Flux

But the control is for `Package Type`. There is no "Flux" packaging format per se, it is a Helm chart.
So packages types would be slightly different from plugin names:
- Helm
- Kapp (or Carvel)
- Helm via Flux

WDYT?

**Review by castelblanque — APPROVED** [review]
LGTM, my only concern is about the "Package types" labels.
Thanks!

File: dashboard/src/components/Catalog/Catalog.tsx (hunk 1 above)
Comment (antgamdia) [reply dashboard/src/components/Catalog/Catalog.tsx:242]: Yes, currently all the plugins are using this `packaging` prefix, so it's safe to use it here (but it's an ugly workaround, though).
As we are adding new plugins (thinking now of the RBAC one we discussed), we may want to add a `type` field, good point; we should do it.

File: dashboard/src/components/Catalog/Catalog.tsx (hunk 2 above)
Comment (antgamdia) [reply dashboard/src/components/Catalog/Catalog.tsx:501]: Good idea. Initially I used "Packaging Plugin" as the filter name, hence the names I chose. But, 100% agree, now we're using "Package Type", we better use this existing function:

https://github.com/vmware-tanzu/kubeapps/blob/88030f61605dab0fd176ae14d4ef22b2b29438a6/dashboard/src/shared/utils.ts#L125-L133

See the result:

image

**Pull request merged**